packages/adminrouter/extra/src/test-harness/tests/test_generic.config.yml

```diff
@@ -74,6 +74,18 @@ endpoint_tests:
     type:
       - master
   - tests:
+      # This test checks whether AR's `/service` endpoint is able to route to
+      # the correct service even though the request url contains duplicate
+      # slashes.  Duplicate slashes is AFAIK not a valid RFC behaviour, but
```

Comment:
I just went a little bit deeper, it turns out that deploying Marathon Service into Application ID `foo/bar///baz/nginx` resulted in `/marathon/v2/apps` output:
```
{

    "apps": [
        {
            "id": "/foo/bar/baz/nginx",
            "acceptedResourceRoles": [
                "*"
            ],
            "backoffFactor": 1.15,
            "backoffSeconds": 1,
            "container": {
                "type": "DOCKER",
                "docker": {
                    "forcePullImage": false,
                    "image": "bitnami/nginx:1.10.3-r1",
                    "parameters": [ ],
                    "privileged": false
                },
                "volumes": [ ],
                "portMappings": [
                    {
                        "containerPort": 80,
                        "hostPort": 0,
                        "labels": { },
                        "protocol": "tcp",
                        "servicePort": 10000
                    },
                    {
                        "containerPort": 443,
                        "hostPort": 0,
                        "labels": { },
                        "protocol": "tcp",
                        "servicePort": 10001
                    }
                ]
            },
```

So internally, Marathon also normalizes Application ID/removes slashes and current AR behaviour is consistent with Marathon's.

Thanks @adamtheturtle for going so deep and for being an awsome reviewer, this was an interesting finding.